packages/web3-core-method/src/index.js

```diff
@@ -653,7 +653,15 @@ Method.prototype.buildCall = function () {
                 if (!err && method.isRevertReasonString(result)){
                     reasonData = result.substring(10);
                 } else if (err && err.data){
-                    reasonData = err.data.substring(10);
+                    // workaround embedded error details got from some providers like MetaMask
+                    if (typeof err.data === 'object') {
```

Comment:
@nicos99 Thanks for your contribution. Can you also add a unit test for it?

Reply:
Yes @jdevcs, it's done!